pkg/stanza/flush/flush.go

```diff
@@ -52,8 +52,8 @@ func (s *State) Func(splitFunc bufio.SplitFunc, period time.Duration) bufio.Spli
 			return 0, nil, nil
 		}
 
-		// Flush timed out
-		if time.Since(s.LastDataChange) > period {
+		// EOF reached, flushing once timer expires
```

Comment:
I think @ChrsMark's comment [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32170#issuecomment-2040710725) is worth incorporating as well, if you agree.